score-lang-content-verifier/src/main/java/org/openscore/lang/tools/verifier/VerifierHelper.java

```diff
@@ -0,0 +1,134 @@
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.log4j.Logger;
+import org.openscore.lang.compiler.SlangCompiler;
+import org.openscore.lang.compiler.modeller.model.Executable;
+import org.openscore.lang.compiler.modeller.model.SlangFileType;
+import org.openscore.lang.compiler.scorecompiler.ScoreCompiler;
+import org.openscore.lang.entities.CompilationArtifact;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.openscore.lang.compiler.SlangSource.fromFile;
+
+/*
+ * Created by stoneo on 2/9/2015.
+ */
+
+/**
+ * Verifies all files with extensions: .sl, .sl.yaml or .sl.yml in a given directory are valid
+ */
+public class VerifierHelper {
+
+    @Autowired
+    private SlangCompiler slangCompiler;
+
+    @Autowired
+    private ScoreCompiler scoreCompiler;
+
+    private final static Logger log = Logger.getLogger(SlangFilesVerifier.class);
+
+    private String[] SLANG_FILE_EXTENSIONS = {"sl", "sl.yaml", "sl.yml"};
+
+    private Map<String, Executable> slangModels = new HashMap<>();
+
+    private Map<String, CompilationArtifact> compiledArtifacts = new HashMap<>();
+
+    /**
+     * Transform all Slang files in given directory to Slang models, and store them
+     * @param directoryPath given directory containing all Slang files
+     */
+    public void verifyAllSlangFilesInDirAreValid(String directoryPath){
+        transformSlangFilesInDirToModels(directoryPath);
+        compileAllSlangModels();
+    }
+
+    private void transformSlangFilesInDirToModels(String directoryPath) {
+        Validate.notNull(directoryPath, "Directory path path can not be null");
+        Collection<File> slangFiles = FileUtils.listFiles(new File(directoryPath), SLANG_FILE_EXTENSIONS, true);
+        log.info("Start compiling all slang files under: " + directoryPath);
+        log.info(slangFiles.size() + " .sl files were found");
+        for(File slangFile: slangFiles){
+            Validate.isTrue(slangFile.isFile(), "filePath must lead to a file");
+            Executable sourceModel;
+            try {
+                sourceModel = slangCompiler.preCompile(fromFile(slangFile));
+            } catch (Exception e) {
+                log.error("Failed creating Slang ,models for directory: " + directoryPath + ". Exception is : " + e.getMessage());
+                throw e;
+            }
+            verifyStaticCode(slangFile, sourceModel);
+            slangModels.put(getUniqueName(sourceModel), sourceModel);
+        }
+        if(slangFiles.size() != slangModels.size()){
+            throw new RuntimeException("We found: " + slangFiles.size() + " .sl files in path: " + directoryPath + ". We managed to create slang models for only: " + slangModels.size());
```

Comment:
@orius123 If for some reason the precompile() returned null, and did not throw an exception (as was the case for the system props), I think we want to get an error that not all files were compiled